rama-http/src/headers/x_robots_tag_components/robots_tag_builder.rs

```diff
@@ -0,0 +1,70 @@
+use crate::headers::util::value_string::HeaderValueString;
+use crate::headers::x_robots_tag_components::custom_rule::CustomRule;
+use crate::headers::x_robots_tag_components::max_image_preview_setting::MaxImagePreviewSetting;
+use crate::headers::x_robots_tag_components::robots_tag::RobotsTag;
+use crate::headers::x_robots_tag_components::valid_date::ValidDate;
+
+macro_rules! builder_field {
+    ($field:ident, $type:ty) => {
+        paste::paste! {
+            pub(super) fn [<$field>](mut self, [<$field>]: $type) -> Self {
+                self.0.[<set_ $field>]([<$field>]);
+                self
+            }
+
+            pub(super) fn [<set_ $field>](&mut self, [<$field>]: $type) -> &mut Self {
+                self.0.[<set_ $field>]([<$field>]);
+                self
+            }
+
+            pub(super) fn [<with_ $field>](mut self, [<$field>]: $type) -> Self {
+                self.0.[<set_ $field>]([<$field>]);
+                self
+            }
+        }
+    };
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(super) struct RobotsTagBuilder<T = ()>(T);
+
+impl RobotsTagBuilder<()> {
+    pub(super) fn new() -> Self {
+        RobotsTagBuilder(())
+    }
+
+    pub(super) fn bot_name(
+        self,
+        bot_name: Option<HeaderValueString>,
+    ) -> RobotsTagBuilder<RobotsTag> {
```

Comment:
this is still not enough to allow it to be build though as no tag is defined yet and AFAIK a robots-tag with just a bot name is non-sensical.